mm2src/mm2_main/src/mm2.rs

```diff
@@ -160,10 +160,30 @@ pub async fn lp_main(
         .with_datetime(datetime.clone())
         .into_mm_arc();
     ctx_cb(try_s!(ctx.ffi_handle()));
+
+    #[cfg(not(target_arch = "wasm32"))]
+    spawn_ctrl_c_handler(ctx.clone());
+
     try_s!(lp_init(ctx, version, datetime).await);
     Ok(())
 }
 
+/// Handles CTRL-C signals and shutdowns the KDF runtime gracefully.
+///
+/// It's important to spawn this task as soon as `Ctx` is in the correct state.
+#[cfg(not(target_arch = "wasm32"))]
+fn spawn_ctrl_c_handler(ctx: mm2_core::mm_ctx::MmArc) {
+    common::executor::spawn(async move {
+        tokio::signal::ctrl_c()
+            .await
+            .expect("Couldn't listen for the CTRL-C signal.");
+
+        log::info!("Wrapping things up and shutting down...");
+
+        ctx.stop().await.expect("Couldn't stop the KDF runtime.");
+    });
```

Comment:
Shouldn't we handle the errors here just for thoroughness
```suggestion
    common::executor::spawn(async move {
        loop {
            if let Err(e) = tokio::signal::ctrl_c().await {
                log::error!("Error listening for the CTRL-C signal: {}", e);
                continue; // Continue the loop to listen for the signal again
            }

            log::info!("Wrapping things up and shutting down...");

            if let Err(e) = ctx.stop().await {
                log::error!("Error stopping the KDF runtime: {}", e);
                continue; // Continue the loop to listen for the signal again
            }
            break; // Exit the loop if shutdown is successful
        }
    });
```

Reply:
In most cases listening for the signal shouldn't panic. But if there is an OS-related problem this will continuously spin and will exhaust the system resources.

The following logic with `if let Err(e) = ctx.stop().await {` part will require user to trigger `CTRL+C` again in order to run the cleaning logic again. If there is a bug inside the `ctx.stop`, user will not be able to stop the runtime with `SIGINT` signals.